gateway-ha/src/main/java/io/trino/gateway/ha/router/RoutingRulesManager.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.router;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLParser;
+import com.google.common.collect.ImmutableList;
+import io.trino.gateway.ha.config.RoutingRulesConfiguration;
+import io.trino.gateway.ha.domain.RoutingRules;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public final class RoutingRulesManager
```

Comment:
I'm not sure why this class is a static helper class. Could you consider changing to a service class?

Reply:
<img width="850" alt="Screenshot 2024-10-28 at 7 22 30 PM" src="https://github.com/user-attachments/assets/aa287e77-1ab6-4e28-8d3b-ac679ee206a0">

So I was trying to implement these comments and then I got an error saying I need to make the class final as well to have all static methods. Maybe I misunderstood what you meant?

Reply:
@ebyhr any inputs?

Reply:
Please ignore the previous comment and change to a service class.